tar/ccache/ccache_read.c

```diff
@@ -65,7 +68,17 @@ read_rec(void * cookie)
 	ccr->ino = (ino_t)le64dec(ccre.ino);
 	ccr->size = (off_t)le64dec(ccre.size);
 	ccr->mtime = (time_t)le64dec(ccre.mtime);
+#if SIZE_MAX < UINT64_MAX
+	ccr_nch = le64dec(ccre.nch);
```

Comment:
Technically using a large cache from a 64-bit system on a 32-bit system could trigger the "Cache file is corrupt" message, yes... but that's far less likely than the cache actually getting corrupted, so I'd prefer to have a false-positive warning in a very unusual use case than fail to warn about something which is probably a problem.

... especially since the solution in both cases is the same.